### PR TITLE
PR - Codebase modification: Changing visibility property in ConnectorViewModel to work with `IsCollapsed` from groups

### PR DESCRIPTION
**Pull request by M-JULIANI**
@QilongTang please make sure https://github.com/DynamoDS/Dynamo/pull/12039 and https://github.com/DynamoDS/Dynamo/pull/12040 (in that order) are merged to master prior to this PR being merged.

### Description
This PR changes the `IsVisible` property on the `ConnectorModel`, `ConnectorViewModel`, `ConnectorPinViewModel` and related classes to the inverse boolean `IsCollapsed`. This has repercussions across the codebase as you can see from the number of files changed.

*This PR stops short of ensuring that all pins work seamlessly with groups. Annotation models do _not_ contain pins, and thus pins don't behave as expected other notes and nodes do (with groups) currently. That is pins don't currently move with groups.
This can be tackled in a separate PR:  the thought is the current PR is about bringing the codebase up-to-date with the inverse logic change and ensuring that the change doesn't break anything. (I ran some unit tests but would not be surprised if there are regressions that I didn't see in the unit tests I ran).

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Reviewers
@QilongTang 

### FYIs
@SHKnudsen 
@Amoursol

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/Core/ConnectorPinViewModel.cs [hunk 1]
```diff
@@ -138,39 +138,42 @@ public bool IsSelected
...
-                RaisePropertyChanged(nameof(IsVisible));
-                RaisePropertyChanged(nameof(IsPartlyVisible));
+                if (isCollapsed == value)
+                {
+                    return;
+                }
+
+                isCollapsed = value;
+                RaisePropertyChanged(nameof(IsCollapsed));
+                RaisePropertyChanged(nameof(IsTemporarilyVisible));
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/ConnectorPinViewModel.cs:155]: Can you remind me why we need to raise change here if the value is not touched?

File: src/DynamoCoreWpf/ViewModels/Core/ConnectorPinViewModel.cs [hunk 2]
```diff
@@ -138,39 +138,42 @@ public bool IsSelected
...
+                return isTemporarilyVisible;
             }
             set
             {
-                isPartlyVisible = value;
-                RaisePropertyChanged(nameof(IsPartlyVisible));
-                RaisePropertyChanged(nameof(IsVisible));
+                isTemporarilyVisible = value;
+                RaisePropertyChanged(nameof(IsTemporarilyVisible));
+                RaisePropertyChanged(nameof(IsCollapsed));
```
Comment (QilongTang) [inline src/DynamoCoreWpf/ViewModels/Core/ConnectorPinViewModel.cs:176]: Same comment here

**Review by QilongTang — COMMENTED** [review]
Two comments then LGTM

File: src/DynamoCoreWpf/ViewModels/Core/ConnectorPinViewModel.cs (hunk 1 above)
Comment (M-JULIANI) [reply src/DynamoCoreWpf/ViewModels/Core/ConnectorPinViewModel.cs]: Fixed- this was a leftover.

File: src/DynamoCoreWpf/ViewModels/Core/ConnectorPinViewModel.cs (hunk 2 above)
Comment (M-JULIANI) [reply src/DynamoCoreWpf/ViewModels/Core/ConnectorPinViewModel.cs]: Fixed- this was a leftover.

**Review by QilongTang — APPROVED** [review]

**Pull request merged**